File: arrays/RemoveElement.java
package arrays;

public class RemoveElement {
    public int removeElement(int[] nums, int val) {
        
        int currentSize = nums.length;

        int count = 0;

        while(count < currentSize){
            if(nums[count] == val){
                nums[count] = nums[currentSize-1];
                currentSize--;
                continue;
            }
            else{
                count++;
            }
        }

        return currentSize;
    }    
}

File: arrays/RemoveElement.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        length = len(nums)
        p = 0

        while p < length:
            if nums[p] == val:
                nums[p] = nums[length-1]
                length = length-1
            else:
                p = p+1
        
        return length        